src/Access/AccessControlManager.cpp

```diff
@@ -419,6 +421,18 @@ std::shared_ptr<const ContextAccess> AccessControlManager::getContextAccess(
     params.http_method = client_info.http_method;
     params.address = client_info.current_address.host();
     params.quota_key = client_info.quota_key;
+
+    /// Extract the last entry from comma separated list of X-Forwarded-For addresses.
+    /// Only the last proxy can be trusted (if any).
```

Comment:
Why only the last proxy can be trusted?

Reply:
Only the last proxy can be trusted, because we can limit access from that proxy with the list of IP networks.
Otherwise user can manually send request with any X-Forwarded-For.

For example, I will limit the list of IP networks for user with this: https://www.cloudflare.com/ips/ and then make quota keyed by X-Forwarded-For.